classes/shop/Shop.php

```diff
@@ -464,13 +465,36 @@ public function getAddress()
     }
 
     /**
-     * Get shop theme name
-     *
-     * @return string
+     * Set shop theme details from Json data
      */
+    public function setTheme()
+    {
+        $dir = _PS_ALL_THEMES_DIR_.$this->theme_name;
+        $theme_data = Yaml::parse(file_get_contents(
```

Comment:
Hmm, the [doc](http://api.symfony.com/3.0/Symfony/Component/Yaml/Yaml.html#method_parse) doesn't say it is deprecated, where did you find this information?